subprojects/platform-base/src/main/java/org/gradle/language/base/internal/tasks/SimpleStaleClassCleaner.java

```diff
@@ -31,7 +31,7 @@ public SimpleStaleClassCleaner(TaskOutputsInternal taskOutputs) {
     public void execute() {
         String prefix = getDestinationDir().getAbsolutePath() + File.separator;
         for (File f : taskOutputs.getPreviousOutputFiles()) {
-            if (f.getAbsolutePath().startsWith(prefix)) {
+            if (f.getAbsolutePath().startsWith(prefix) && f.isFile()) {
```

Comment:
Moreover, we would need to make sure to not delete the destination directory, too.